src/main/java/ch/njol/skript/lang/SyntaxElement.java

```diff
@@ -47,4 +51,18 @@ default ParserInstance getParser() {
 		return ParserInstance.get();
 	}
 
+	/**
+	 * Available during parsing to check whether a script has an experiment enabled.
+	 * @param experiment The experiment to test.
+	 * @return Whether the currently-parsing script
+	 * @throws SkriptAPIException If the current script is not available in the parser instance.
+	 */
+	@Override
+	default boolean hasExperiment(Experiment experiment) throws SkriptAPIException {
```

Comment:
I'd prefer to avoid integrating Experiments (or any feature really) so deeply with the core classes. I do not think it is much trouble to use `getParser().getCurrentScript().hasExperiment(...)` (we do the same for events)